daemon/src/maker_inc_connections.rs

```diff
@@ -441,6 +377,119 @@ impl Actor {
     }
 }
 
+/// Sets up a new connection on the given stream including all relevant background tasks like
+/// heartbeat and reading messages from the stream.
+async fn setup_new_connection(
+    this: Address<Actor>,
+    noise_priv_key: x25519_dalek::StaticSecret,
+    heartbeat_interval: Duration,
+    stream: TcpStream,
+    address: SocketAddr,
+) {
+    let (mut read, write, taker_id) = match upgrade(stream, noise_priv_key).await {
+        Ok((read, write, identity)) => (read, write, identity),
+        Err(e) => {
+            tracing::warn!(address = %address, "Failed to upgrade incoming connection: {:#}", e);
+            return;
+        }
+    };
+
```

Comment:
Can we have a log here telling that a new taker connected, e.g. 
```
tracing::info!(%taker_id, address = %taker_address, "New taker connected");
```

Could be useful to see because the next log will only happen if the connection was upgraded

Reply:
How is that log different from the one in `upgrade`?

```rust
tracing::info!(taker_id = %taker_id, %taker_address, "Connection upgrade successful");
```

Reply:
If I understand it correctly then the one in `upgrade` is only printed if the handshake was successful, while logging here would should show us all connection attempts.

Reply:
I am not sure I follow.

- "Here" as to where the comment is made is after `upgrade` completes _successfully_ so that is already logged inside `upgrade`.
- Your log suggests to log `taker_id` but that is only available after an upgrade. Noise needs to complete the handshake to learn the other party's identity. That is what `upgrade` does. We could log between the noise and the `Hello` handshake but I am not sure that is worth it?
- There is a log inside `upgrade` that logs for every connection attempt. It is on line 445.